### ISSUE TITLE
unable to compose ostree-based disk images from ostree repo with multiple commits

### ISSUE THREAD

**Issue by miabbott** [osbuild/osbuild-composer#3644]
**Describe the bug**

When composing an `ostree`-based disk image (i.e. `edge-raw-image`, `edge-installer`) from an ostree repo that has multiple commits, the compose fails.

**Environment**
 * OS version (`/etc/os-release` and `/etc/redhat-release`):

```
$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="9.2 (Plow)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="9.2"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Red Hat Enterprise Linux 9.2 (Plow)"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:redhat:enterprise_linux:9::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_BUGZILLA_PRODUCT_VERSION=9.2
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.2"
```

 * osbuild-composer version (`rpm -qi osbuild-composer)`

```
$ rpm -qi osbuild-composer
Name        : osbuild-composer
Version     : 76
Release     : 2.el9_2.2
Architecture: x86_64
Install Date: Tue Aug 22 14:48:52 2023
Group       : Unspecified
Size        : 17587
License     : Apache-2.0
Signature   : RSA/SHA256, Wed Apr 26 06:04:11 2023, Key ID 199e2f91fd431d51
Source RPM  : osbuild-composer-76-2.el9_2.2.src.rpm
Build Date  : Wed Apr 26 04:29:29 2023
Build Host  : x86-64-02.build.eng.rdu2.redhat.com
Packager    : Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>
Vendor      : Red Hat, Inc.
URL         : https://github.com/osbuild/osbuild-composer
Summary     : An image building service based on osbuild
Description :

A service for building customized OS artifacts, such as VM images and OSTree
commits, that uses osbuild under the hood. Besides building images for local
usage, it can also upload images directly to cloud.

It is compatible with composer-cli and cockpit-composer clients.
```

**To Reproduce**

1. Create empty blueprint in order to build `edge-commit`
2. Start `edge-commit` compose:  `composer-cli compose start-ostree <blueprin> edge-commit`
3. Grab tarball and explode to disk
4. Copy `repo` directory to `/usr/share/nginx/html` (or similar) to be served via HTTP: `cp -r ./repo /usr/share/nginx/html/`
5. Modify existing blueprint to have additional package, in order to build new ostree commit
6. Start new `edge-commit` compose
7. Grab new tarball, use `ostree pull-local` to copy new commit to existing ostree repo:  `ostree --repo=/usr/share/nginx/html/repo pull-local ./repo`
8. Create empty blueprint to compose disk image
9. Start `edge-raw-image`/`edge-installer` compose: `composer-cli compose start-ostree --ref rhel/9/x86_64/edge --url http://192.168.122.150/repo <blueprint> edge-installer`

**Expected behavior**

Disk image compose is successful

**Additional context**

The error in the journal looks like it is failing on an internal use of `ostree pull-local` trying to pull data into a temporary ostree repo:

```
Aug 22 16:17:08 localhost.localdomain osbuild-composer[5105]: 2023/08/22 16:17:08 POST /api/v1/compose
Aug 22 16:17:15 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:17:15-04:00" level=info msg="Running job 'b5b21628-80c5-4670-9027-422b37d4df1b' (osbuild)\n"
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]: Traceback (most recent call last):
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/bin/osbuild", line 33, in <module>
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     sys.exit(load_entry_point('osbuild==81', 'console_scripts', 'osbuild')())
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/main_cli.py", line 165, in osbuild_cli
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     r = manifest.build(
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/pipeline.py", line 467, in build
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     res = pl.run(store, monitor, libdir, stage_timeout)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/pipeline.py", line 371, in run
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     results = self.build_stages(store,
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/pipeline.py", line 344, in build_stages
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     r = stage.run(tree,
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/pipeline.py", line 212, in run
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     data = ipmgr.map(ip)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/inputs.py", line 91, in map
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     reply = client.call("map", args)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/host.py", line 348, in call
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     ret, _ = self.call_with_fds(method, args)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/host.py", line 384, in call_with_fds
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     raise error
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]: osbuild.host.RemoteError: CalledProcessError: Command '['ostree', 'pull-local', '/var/cache/osbuild-worker/osbuild-store/sources/org.osbuild.ostree/repo', 'd289203132531644640301eb930549e3f19aae2dc0067908b2acb65b2f41e21e', '--repo=/var/cache/osbuild-worker/
osbuild-store/tmp/buildroot-tmp-mtwe1add/inputs/commits/repo']' returned non-zero exit status 1.
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:    File "/usr/lib/python3.9/site-packages/osbuild/host.py", line 268, in serve
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     reply, reply_fds = self._handle_message(msg, fds)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/host.py", line 301, in _handle_message
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     ret, fds = self.dispatch(name, args, fds)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/python3.9/site-packages/osbuild/inputs.py", line 121, in dispatch
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     r = self.map(store,
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/osbuild/inputs/org.osbuild.ostree", line 141, in map
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     reply = export(refs, source, target)
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/osbuild/inputs/org.osbuild.ostree", line 108, in export
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     ostree("pull-local", repo_cache, commit,
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib/osbuild/inputs/org.osbuild.ostree", line 90, in ostree
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     subprocess.run(["ostree"] + args,
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:   File "/usr/lib64/python3.9/subprocess.py", line 528, in run
Aug 22 16:17:47 localhost.localdomain osbuild-worker[87077]:     raise CalledProcessError(retcode, process.args,
Aug 22 16:17:47 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:17:47-04:00" level=error msg="osbuild job failed: osbuild build failed" jobId=b5b21628-80c5-4670-9027-422b37d4df1b
Aug 22 16:17:47 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:17:47-04:00" level=warning msg="Job 'b5b21628-80c5-4670-9027-422b37d4df1b' (osbuild) failed: error decoding osbuild output: unexpected end of JSON input\nthe raw output:\n"
```

If I try to compose an ostree-based disk image from a repo with only a single commit (i.e. after step 4 above), it is successful.

**Comment by miabbott** [comment]
Successful compose:

<details>

```
Aug 22 16:11:08 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:11:08-04:00" level=info msg="Running job 'a02eedd8-dca5-4cc3-878a-6450596652b6' (osbuild)\n"
Aug 22 16:11:48 localhost.localdomain ostree[85491]: libostree pull from '74e00493-9967-44f8-b781-1739d3b41ffb' for 0 refs complete
                                                     security: GPG: disabled 
                                                     security: SIGN: disabled http: TLS
                                                     non-delta: meta: 3480 content: 26858
                                                     transfer: secs: 39 size: 841.9 MB
Aug 22 16:12:38 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:38 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:38 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 19920863
Aug 22 16:12:38 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:38 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 19920863
Aug 22 16:12:39 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:39 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 19920863
Aug 22 16:12:39 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:39 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 260096
Aug 22 16:12:40 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:40 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 786432
Aug 22 16:12:40 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:40 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 19920863
Aug 22 16:12:41 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:41 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 786432
Aug 22 16:12:41 localhost.localdomain kernel: loop1: detected capacity change from 0 to 20971520
Aug 22 16:12:41 localhost.localdomain kernel: loop1: detected capacity change from 20971520 to 260096
Aug 22 16:12:41 localhost.localdomain kernel: loop2: detected capacity change from 0 to 20971520
Aug 22 16:12:41 localhost.localdomain kernel: loop2: detected capacity change from 20971520 to 19920863
Aug 22 16:12:41 localhost.localdomain kernel: XFS (dm-4): Mounting V5 Filesystem
Aug 22 16:12:41 localhost.localdomain kernel: XFS (dm-4): Ending clean mount
Aug 22 16:12:41 localhost.localdomain kernel: XFS (loop0): Mounting V5 Filesystem
Aug 22 16:12:41 localhost.localdomain kernel: XFS (loop0): Ending clean mount
Aug 22 16:12:48 localhost.localdomain kernel: XFS (loop0): Unmounting Filesystem
Aug 22 16:12:49 localhost.localdomain kernel: XFS (dm-4): Unmounting Filesystem
Aug 22 16:12:49 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:49 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 19920863
Aug 22 16:12:49 localhost.localdomain kernel: loop0: detected capacity change from 0 to 20971520
Aug 22 16:12:49 localhost.localdomain kernel: loop0: detected capacity change from 20971520 to 19920863
Aug 22 16:13:50 localhost.localdomain NetworkManager[764]: <info>  [1692735230.1187] dhcp4 (enp1s0): state changed new lease, address=192.168.122.150
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="build pipeline results:\n" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.rpm success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.selinux success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="image-tree pipeline results:\n" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.init-fs success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.pull success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.os-init success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.mkdir success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.deploy success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.remotes success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.fillvar success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.config success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.fstab success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ignition success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.users success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.keymap success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.locale success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.grub2 success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.ostree.selinux success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="image pipeline results:\n" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.truncate success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.sfdisk success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.luks2.format success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.clevis.luks-bind success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.lvm2.create success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.mkfs.fat success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.mkfs.xfs success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.mkfs.xfs success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.copy success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.lvm2.metadata success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.luks2.remove-key success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.grub2.inst success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="xz pipeline results:\n" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:06 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:06-04:00" level=info msg="  org.osbuild.xz success" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:07 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:07-04:00" level=info msg="osbuild job succeeded" jobId=a02eedd8-dca5-4cc3-878a-6450596652b6
Aug 22 16:15:07 localhost.localdomain osbuild-worker[5106]: time="2023-08-22T16:15:07-04:00" level=info msg="Job 'a02eedd8-dca5-4cc3-878a-6450596652b6' (osbuild) finished"
```

</details>

**Comment by achilleas-k** [comment]
In step 6

> * Start new `edge-commit` compose

is the new commit built as a child of the first one or is it completely separate?

**Comment by miabbott** [comment]
> is the new commit built as a child of the first one or is it completely separate?

The available command history + journal on my RHEL 9.2 system isn't sufficient to determine the exact commands used.

I'll need to experiment some more to see if I can recreate this behavior.

**Comment by miabbott** [comment]
I've not been able to reproduce this with multiple commits in a repo.  I've tried with commits that have proper parent commits and with commits that are "independent".

If I encounter this again, I'll reopen.
